Replace deprecated v13.app.FragmentStatePagerAdapter for support v4 FragmentStateAdapter

mzorz opened the pull request:
While looking into some unrelated issue, I realized that there were several places where the deprecated `v13.app.FragmentStatePagerAdapter` was being used, and in order to isolate and be able to null a possible problem out I decided to do a small refactor and get rid of that v13 implementation, in favor of the currently supported and recommended support.v4 one.

To test:
Compile and make sure these parts of the app are working:
- Notifications (4th app tab to the right) -> tap on a comment notification, and observe the comment detail screen appears and shows the contents of it.
- First tab: tap on Comments -> see the comments list -> tap on a comment and see the comment detail screen appears as well
- Reader: choose a post that has photos in it, and tap on a photo. The photo should be displayed in the photo viewer.

nbradbury commented:
@mzorz Looks good! Do we also need to replace the usages of the v13 `FragmentPagerAdapter` or is that a task for another PR?

<img width="534" alt="screen shot 2018-05-17 at 9 27 04 am" src="https://user-images.githubusercontent.com/3903757/40180360-9f1bd2ac-59b4-11e8-9910-ed9b01c9f91b.png">

mzorz commented:
> Do we also need to replace the usages of the v13 `FragmentPagerAdapter`... ?

- Changed `android.support.v13.app.FragmentPagerAdapter` and replaced with `android.support.v4.app.FragmentPagerAdapter` in fe2b374e46
This affects several sections of the app: Plans, Editor and the Reader. Please let's do some testing on these parts to make sure nothing seems odd.

- I also went an extra mile and removed the use of `FragmentCompat` in a6a627e095 so we don't have any other deprecated v13 usages.
- Effectively removed the references to the v13 support library in bd9b6ac002

To test this last part:
- eliminate the app from your phone, or revoke all permissions (camera / storage specifically)
- edit a new draft
- tap the + icon to bring the photo picker up (this involves already some FragmentCompat usage)
- now, allow the permission request (this tests the other part where FragmentCompat was involved)

Everything should run fine and you should be able to pick a picture and upload it @nbradbury

nbradbury commented:
@mzorz Nice work on this! I tested this on multiple devices (going back to API 16) and everything seems to be working fine. :shipit:
